community/bolt/src/test/java/org/neo4j/bolt/transport/TransportWriteThrottleTest.java

```diff
@@ -166,85 +164,122 @@ public void shouldResumeWhenWritableOnceAgain() throws Exception
         throttle.acquire( channel );
 
         // expect
-        assertThat( lock.lockCallCount(), greaterThan( 0 ) );
-        assertThat( lock.unlockCallCount(), is( 0 ) );
+        verify( lock, atLeast( 1 ) ).lock( any(), anyLong() );
+        verify( lock, never() ).unlock( any() );
     }
 
     @Test
     public void shouldResumeWhenWritabilityChanged() throws Exception
     {
+        TestThrottleLock lockOverride = new TestThrottleLock();
+
         // given
-        TransportThrottle throttle = newThrottleAndInstall( channel );
+        TransportThrottle throttle = newThrottleAndInstall( channel, lockOverride );
         when( channel.isWritable() ).thenReturn( false );
 
-        Future future = Executors.newSingleThreadExecutor().submit( () -> throttle.acquire( channel ) );
+        Thread throttleThread = Executors.defaultThreadFactory().newThread( () -> throttle.acquire( channel ) );
```

Comment:
Maybe use `OtherThreadRule` class? It allows running things in a different thread and has ability to check thread state.

Reply:
Thanks for pointing out, didn't know that.